tsdb/wal/watcher.go

```diff
@@ -466,10 +471,14 @@ func (w *Watcher) readSegment(r *LiveReader, segmentNum int, tail bool) error {
 		samples []record.RefSample
 		send    []record.RefSample
 	)
+
 	for r.Next() && !isClosed(w.quit) {
 		rec := r.Record()
 		w.recordsReadMetric.WithLabelValues(recordType(dec.Type(rec))).Inc()
 
+		// Sets the current Segment files it's reading from
+		w.CurrSegmentFile = SegmentName("", segmentNum)
```

Comment:
this should be set in the same location as the metric is being set

Reply:
would this not go in the readSegment() since this is where the current segment is being read